akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpRequestRendererFactory.scala

```diff
@@ -102,8 +105,16 @@ private[http] class HttpRequestRendererFactory(userAgentHeader: Option[headers.`
     def renderContentLength(contentLength: Long) =
       if (method.isEntityAccepted && (contentLength > 0 || method.requestEntityAcceptance == Expected)) r ~~ `Content-Length` ~~ contentLength ~~ CrLf else r
 
-    def renderStreamed(body: Source[ByteString, Any]): RequestRenderingOutput =
-      RequestRenderingOutput.Streamed(renderByteStrings(r, body))
+    def renderStreamed(body: Source[ByteString, Any]): RequestRenderingOutput = {
+      val headerPart = Source.single(r.get)
+      val stream = ctx.sendEntityTrigger match {
+        case None ⇒ headerPart ++ body
+        case Some(future) ⇒
+          val barrier = Source.fromFuture(future).drop(1).asInstanceOf[Source[ByteString, Any]]
```

Comment:
Not quite, it's used as a `barrier` (think like stream-ish countdownlatch basically).
Once the future completes, it emits an element, which we drop, and the stream completes which then causes the rest of the concated stream to be emitted: `headerPart ++ barrier ++ body`. So barrier never emits anything, however it prevents `body` from being signalled until barrier is triggered.